socs/agents/http_camera/agent.py

```diff
@@ -159,29 +157,36 @@ def acq(self, session, params=None):
                         response = requests.get(url, params=payload, stream=True, timeout=5, verify=False)
                     elif camera['brand'] == 'acti':
                         response = requests.get(url, params=payload, stream=True, timeout=5)
-                    connected = True
                 except (requests.exceptions.RequestException, ReadTimeoutError) as e:
                     self.log.error(f'{e}')
                     self.log.info("Unable to get response from camera.")
-                    connected = False
                     data[camera['location']]['last_attempt'] = time.time()
-                    data[camera['location']]['connected'] = connected
+                    data[camera['location']]['connected'] = False
                     continue
                 self.log.debug("Received screenshot from camera.")
 
                 # Write screenshot to file and update latest file
-                with open(filename, 'wb') as out_file:
-                    shutil.copyfileobj(response.raw, out_file)
-                    # Ensure all data is written to the disk before copying to latest
-                    out_file.flush()
-                    os.fsync(out_file.fileno())
-                self.log.debug(f"Wrote {ctime}.jpg to /{camera['location']}/{ctime_dir}.")
+                try:
+                    with open(filename, 'wb') as out_file:
+                        for chunk in response.iter_content(chunk_size=8192):
+                            if chunk:
+                                out_file.write(chunk)
+                        out_file.flush()
+                        os.fsync(out_file.fileno())
+                    self.log.debug(f"Wrote {ctime}.jpg to /{camera['location']}/{ctime_dir}.")
+                except BaseException as e:
```

Comment:
Catching `BaseException` like this isn't ideal. That catches exceptions that indicate the program should exit like `SystemExit` and `KeyboardInterrupt`, which if raised at the right time could make the agent difficult to stop.

With this code block:
```
shutil.copyfileobj(response.raw, out_file)
# Ensure all data is written to the disk before copying to latest
out_file.flush()
os.fsync(out_file.fileno())
```

We know the error we were running into is another `urllib3.exceptions.ReadTimeoutError` on the `shutil.copyfileobj` line. I don't know that you're going to see that with the new chunk iteration code.

Good practice is to catch and handle specific errors, so my preference here is to `except ReadTimeoutError as e`, like you do above, with the log indicating a timeout occurred. If needed, catching `Exception` is better than the broader `BaseException`, as that doesn't catch those exit exceptions.